### DISCUSSION TITLE
Xiaoxiang / JBD / Overkill BMS Bluetooth Connection

### DISCUSSION THREAD

**Discussion by drageeblue** [theengs/decoder#143] [New devices/things integration]
Hi,

I have this type of BMS. How can we decode this device datas.

I am sharing my screens and log and debug file from nrf connect app.

![image](https://user-images.githubusercontent.com/46354164/177284498-dca5fba9-dd6b-4a71-b8ac-3fa158d53bb2.png)

![image](https://user-images.githubusercontent.com/46354164/177284458-57744dbd-54c1-49b3-b529-e71a596c348b.png)

![image](https://user-images.githubusercontent.com/46354164/177284545-f19ad6a1-17cf-407b-b37a-a7f7cec6c413.png)

![image](https://user-images.githubusercontent.com/46354164/177284580-e7e524f9-ae13-43d1-b57e-1277fcae6bd4.png)

Set Parameters:
![image](https://user-images.githubusercontent.com/46354164/177284614-73d5940c-3453-4921-85d8-342cb41318f3.png)

[rf connect logs.zip](https://github.com/theengs/decoder/files/9044710/rf.connect.logs.zip)

And also I installed Thenngs Add-On on my Home Assistant, I can see some Things like below.

![image](https://user-images.githubusercontent.com/46354164/177286625-9cedf6ff-c6fc-4023-a1b3-1ed6c56c6c18.png)

**Comment by 1technophile** [comment]
That's great, can you share another set of scrernshots with different values ?

**Reply by drageeblue** [reply]
You can find as an attachment all logs and mobile app screenshots. But some it has data changes during runtime on mobile apps, not the set parameters.

[xiaoxiangBMS_NRFConnectLogsAndMobileAppsScreenshots.zip](https://github.com/theengs/decoder/files/9045267/xiaoxiangBMS_NRFConnectLogsAndMobileAppsScreenshots.zip)

**Reply by drageeblue** [reply]
Is the attachment files enough for your analysis?

**Reply by DigiH** [reply]
Hi @drageeblue,

unfortunately the device only broadcasts its MAC address in reverse in the manufacturerdata. All the other information you are interested in can only be gotten by actively connecting to the device, which is out of scope of Decoder, as it can only decode freely broadcast messages.

An alternative option for you to try would be the READ functionality of OpenMQTTGateway, with using the services and characteristics from your nRF Connect tests

https://docs.openmqttgateway.com/use/ble.html#read-write-ble-characteristics-over-mqtt-esp32-only

**Reply by drageeblue** [reply]
Hi DigiH,

I received datas from bms to mqtt->HA via this python code (Sorry for the Turkish object names).

[jbdbms-8-mqtt.zip](https://github.com/theengs/decoder/files/9069306/jbdbms-8-mqtt.zip)

Reference Link: https://github.com/tgalarneau/bms

![image](https://user-images.githubusercontent.com/46354164/177928542-98e0e505-792c-4f2e-b973-ebff14f44db8.png)

**Reply by DigiH** [reply]
Hi drageeblue,

I'm glad you found a solution which connects to your BMS to get all the relevant data 👍
